Multiline regexp - wrong line reported

I'm using this:
```
'regex/invalid': [
	'error',
	[
		{
			regex: '[`\'"][\\s\\n]*\\{[\\s\\n]*"',
			message: 'No inline JSON is allowed, use JSON.stringify() instead',
			replacement: {
				function: 'return text', // make it unfixable
			},
		},
	],
],
```
to catch these:
```
			body: `{
	"TotalItems":5,
	"TotalPages":2,
	"Items":[
		{
			"UserPk":"101",
			"Email":"u1@mail.box",
			"Name":"user1"
		}
	]
}`,
```

Basically, I'm trying to prevent people from writing inline JSON, and use `JSON.stringify()` instead.

If I run it for this example/repro file:
```
const one = `{
	"TotalItems":5,
}`;
const two = `{
	"TotalItems":5,
}`;
const three = `{
	"TotalItems":5,
}`;
```

I'll get:
```
  1:13  error  No inline JSON is allowed, use JSON.stringify() instead  regex/invalid
  5:13  error  No inline JSON is allowed, use JSON.stringify() instead  regex/invalid
  6:15  error  No inline JSON is allowed, use JSON.stringify() instead  regex/invalid
```

But I expect lines 1, 4, and 7 to be reported. Not 1, 5, and 6.

I was able to fix this by keeping track of the full source file and not using `nextLine` to calculate position.
```
function checkRegex(..., originalSource) {
    originalSource = originalSource || source;
    if (sourceDetail.patternIndex !== -1) {
      if (foundDetail.nextLine !== -1) {
        ...
      }
      else {
        report({
          loc: { start: foundStart(originalSource, 1, sourceDetail.patternIndex + rootChar) },
          ...
        })
        checkRegex(..., originalSource)
      }
    }
  }
}
```

--- a/lib/rules/invalid-regex-rule.js
+++ b/lib/rules/invalid-regex-rule.js
@@ -54,7 +54,8 @@ function foundStart(source, nextLine, foundAt) {
     { line: nextLine, column: 0 })
 }
 
-function checkRegex(source, sourceLines, nextLine, nextChar, rootChar, pattern, replace, report) {
+function checkRegex(source, sourceLines, nextLine, nextChar, rootChar, pattern, replace, report, originalSource) {
+  originalSource = originalSource || source
   if (source.length !== 0) {
     const sourceDetail = checkRegexInSource(source, pattern.regex)
     if (sourceDetail.patternIndex !== -1) {
@@ -76,18 +77,19 @@ function checkRegex(source, sourceLines, nextLine, nextChar, rootChar, pattern, 
           fix: replace(rootChar, sourceDetail)
         })
         checkRegex(remainingSource, sourceLines, foundDetail.nextLine, foundDetail.nextChar, rootChar + remainingSourceStart,
-          pattern, replace, report)
+          pattern, replace, report, originalSource)
       }
       else {
         report({
-          loc: { start: foundStart(source, nextLine + 1, sourceDetail.patternIndex) },
+          loc: { start: foundStart(originalSource, 1, sourceDetail.patternIndex + rootChar) },
           message: formatReportMessage(
             pattern,
             from => `Invalid regular expression ${from} found`
           ),
           fix: replace(rootChar, sourceDetail)
         })
-        checkRegex(remainingSource, sourceLines, nextLine + 1, nextChar, rootChar + remainingSourceStart, pattern, replace, report)
+        checkRegex(remainingSource, sourceLines, nextLine + 1, nextChar, rootChar + remainingSourceStart, pattern, replace, report,
+          originalSource)
       }
     }
   }

--- a/tests/lib/rules/invalid-regex-rule.e2e-test.js
+++ b/tests/lib/rules/invalid-regex-rule.e2e-test.js
@@ -146,6 +146,37 @@ const shouldFoundAcrossMultilineInTheMiddleOfLine = {
   }]
 }
 
+const shouldFoundAcrossMultilineInTheMiddleOfLineMultipleTimes = {
+  code: [
+    'const one = `{',
+    '    "TotalItems":5,',
+    '}`;',
+    'const two = `{',
+    '    "TotalItems":5,',
+    '}`;',
+    'const three = `{',
+    '    "TotalItems":5,',
+    '}`;'
+  ].join('\n'),
+  filename: 'some.js',
+  options: [
+    ['[`\'"][\\s\\n]*\\{[\\s\\n]*"']
+  ],
+  errors: [{
+    message: 'Invalid regular expression /[`\'"][\\s\\n]*\\{[\\s\\n]*"/gm found',
+    line: 1,
+    column: 13
+  }, {
+    message: 'Invalid regular expression /[`\'"][\\s\\n]*\\{[\\s\\n]*"/gm found',
+    line: 4,
+    column: 13
+  }, {
+    message: 'Invalid regular expression /[`\'"][\\s\\n]*\\{[\\s\\n]*"/gm found',
+    line: 7,
+    column: 15
+  }]
+}
+
 ruleTester.run(
   'invalid',
   invalidRegexRule, {
@@ -161,7 +192,8 @@ ruleTester.run(
       shouldFoundSameMultiline,
       shouldFoundAcrossLines,
       shouldFoundAcrossMultiline,
-      shouldFoundAcrossMultilineInTheMiddleOfLine
+      shouldFoundAcrossMultilineInTheMiddleOfLine,
+      shouldFoundAcrossMultilineInTheMiddleOfLineMultipleTimes
     ]
   }
 )

